stub/storage.go

```diff
@@ -160,45 +162,96 @@ func renderFieldAsString(fields map[string]interface{}) string {
 	return template
 }
 
-func equals(input1, input2 map[string]interface{}) bool {
-	return reflect.DeepEqual(input1, input2)
+func deepEqual(expect, actual interface{}) bool {
+	return reflect.DeepEqual(expect, actual)
+}
+
+func regexMatch(expect, actual interface{}) bool {
+	match, err := regexp.Match(expect.(string), []byte(actual.(string)))
+	if err != nil {
+		log.Printf("Error on matching regex %s with %s error:%v\n", expect, actual, err)
+	}
+	return match
+}
+
+func equals(expect, actual map[string]interface{}) bool {
+	return find(expect, actual, true, true, deepEqual)
 }
 
 func contains(expect, actual map[string]interface{}) bool {
-	for key, val := range expect {
-		actualvalue, ok := actual[key]
-		if !ok {
-			return ok
+	return find(expect, actual, true, false, deepEqual)
+}
+
+func matches(expect, actual map[string]interface{}) bool {
+	return find(expect, actual, true, false, regexMatch)
+}
+
+func find(expect, actual interface{}, acc, exactMatch bool, f matchFunc) bool {
```

Comment:
Created a recursive function that accepts "expect" and "actual" with any type, an accumulator for nested fields, exactMatch flag for deep equality or contains/reges, and lastly a function parameter that can be "deepEqual" or "matchRegex".

Reply:
if `acc` is always provided with true value, why do we need it anyway?